src/components/picker/index.tsx

```diff
@@ -114,7 +116,7 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
     onChange,
     pickerExpandableRef: pickerExpandable,
     getItemValue,
-    topBarProps,
+    headerProps: type.modal && componentProps.headerProps,
```

Comment:
headerProps is a dialog prop as well, why it not passed in that case?

Reply:
Dialog doesn't uses `cancelSelect` when dismiss, Modal does.
We can pass `headerProps` in each case it won't make any change.

Reply:
So how about we pass 'onCancel' instead of the whole headerProps? This way you won't need to use the deprecated topBarProps in the hook. The component will pass the proper callback or undefined

Reply:
I'll add the `onCancel` prop, great idea.